src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNet.java

```diff
@@ -250,12 +251,16 @@ public void tick(Block b) {
      * @return The frequency of the given node
      */
     private static int getFrequency(Location node) {
-        try {
-            String str = BlockStorage.getLocationInfo(node).getString("frequency");
-            return str == null ? 0 : Integer.parseInt(str);
-        } catch (Exception x) {
-            SlimefunPlugin.logger().log(Level.SEVERE, x, () -> "An Error occurred while parsing a Cargo Node Frequency (" + node.getWorld().getName() + " - " + node.getBlockX() + "," + node.getBlockY() + "," + +node.getBlockZ() + ")");
+
+        String str = BlockStorage.getLocationInfo(node).getString("frequency");
+        if (str == null) {
+            return 0;
+        }
+        if (!PatternUtils.NUMERIC.matcher(str).matches()) {
+            SlimefunPlugin.logger().log(Level.SEVERE, () -> "An Error occurred while parsing a Cargo Node Frequency (" + node.getWorld().getName() + " - " + node.getBlockX() + "," + node.getBlockY() + "," + +node.getBlockZ() + ")");
```

Comment:
```suggestion
            SlimefunPlugin.logger().log(Level.SEVERE, () -> "An Error occurred while parsing a Cargo Node Frequency (" + node.getWorld().getName() + " - " + node.getBlockX() + ',' + node.getBlockY() + ',' + node.getBlockZ() + ')');
```

Reply:
Is there a reason why chars shpuld be used instead? I was just keeping the old error logging.

Reply:
Haha! Yes! Thanks for asking. Chars are faster and use slightly less memory when doing this concatination. When you do "," you're making a new immutable String object which is heavier than just a single char. You're also maintaining an array (byte array or char array depending on version) and hash. Whereas char just holds it's value.

There is a good StackOverflow answer about this too: https://stackoverflow.com/a/24860958

Both are still very quick but if we can use a char, we should.

Reply:
Fair enough, thanks for the insight!